easybuild/scripts/eb_sandbox.sh

```diff
@@ -0,0 +1,287 @@
+                eval $wget_cmd
+                tar xfz $tarball
+            else
+                error "Neither curl nor wget are available, giving up"
+            fi
+        fi
+        mv easybuild-${repo}-${branch} easybuild-${repo}
+    fi
+    if [ -d $sandbox/easybuild-$repo ]
+    then
+        debug "Downloaded easybuild-$repo/$user:$branch to $sandbox/easybuild-$repo"
+    else
+        error "Failed to find $sandbox/easybuild-$repo, some failure was not detected?"
+    fi
+}
+
+# fetch EasyBuild repository according to provided specifications (user:branch or PR)
+fetch() {
+    repo=$1
+    spec=$2
+    if [[ "$spec" =~ ^[0-9][0-9]*$ ]]
+    then
+        # if a pull request number was specified, figure out the user:branch label first
+        pr=$spec
+        debug "Detected PR #$pr for easybuild-$repo repository"
+        pr_data=`curl -sS $GITHUB_API_URL/easybuild-$repo/pulls/$pr`
+        spec=`echo $pr_data | tr ',' '\n' | grep '"label":' | head -1 | sed 's/.*"label": "\([^"]*\)".*/\1/g'`
+        debug "Obtained user:branch $spec for $repo PR #$pr"
+    else
+        # if the specification doesn't match a PR #, assume it's a user:branch label
+        debug "Assuming '${spec}' is an easybuild-$repo repository user:branch label (since it's not a PR #)"
+    fi
+    user=`echo $spec | cut -f1 -d:`
+    branch=`echo $spec | cut -f2 -d:`
+    download_repo $user $repo $branch
+}
+
+# generate a random alphanumeric string of specified length
+random_str() {
+    n=$1
+    # compose list of available characters
+    i=0
+    for c in {a..z} {A..Z} {0..9}
+    do
+        chars[$i]=${c}
+        i=$(($i+1))
+    done
+    # compose random string
+    str=""
+    for i in `seq 1 $n`;
+    do
+        i=$(($RANDOM%${#chars[*]}))
+        str="${str}${chars[$i]}"
+    done
+    echo $str
+}
+
+########
+# MAIN #
+########
+
+# parse command line options
+while getopts ":b:c:df:ghkt:" o; do
+    case "${o}" in
+        b)
+            easyblocks_spec=${OPTARG}
+            ;;
+        c)
+            easyconfigs_spec=${OPTARG}
+            ;;
+        d)
+            debug=true
+            ;;
+        f)
+            framework_spec=${OPTARG}
+            ;;
+        g)
+            use_git=true
+            ;;
+        h)
+            help
+            exit 0
+            ;;
+        k)
+            keep_sandbox=true;
+            ;;
+        t)
+            sandbox_dir=${OPTARG}
+            ;;
+        :)
+            error "Option -$OPTARG requires an argument."
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+eb_args=$*
+extra_eb_args=''
+
+if [ -z $sandbox_dir ]
+then
+    sandbox_dir=$DEFAULT_SANDBOX_DIR
+fi
+
+if [ -z ${framework_spec} ]
+then
+    framework_spec=$DEFAULT_USER_BRANCH
+fi
+if [ -z ${easyblocks_spec} ]
+then
+    easyblocks_spec=$DEFAULT_USER_BRANCH
+fi
+if [ -z ${easyconfigs_spec} ]
+then
+    easyconfigs_spec=$DEFAULT_USER_BRANCH
```

Comment:
This should change to: `if [[ -z ${easyconfigs_spec} || "$easyconfigs_spec" =~ ^[0-9][0-9]*$ ]]`.
If you specify a PR, it should just be passed on to `--from-pr`. The branch on which it is executed should always be master or develop. Now you can have the situation that a run fails because a certain easyconfig is merged in develop after the PR opened, leading to unresolved deps.